Remove redundant fallback for view_as_complex

int3 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #103261

This enables lowering to work for it

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @aakhundov

Differential Revision: [D46585029](https://our.internmc.facebook.com/intern/diff/D46585029)

int3 committed:
Remove redundant fallback for view_as_complex

This enables lowering to work for it

ghstack-source-id: 0e76ff58265ce92ec3888f7a41487ae71dbbdc6c
Pull Request resolved: https://github.com/pytorch/pytorch/pull/103261

desertfire reviewed (approved):
Thanks for fixing it.

eellison commented on torch/_inductor/lowering.py:
```diff
@@ -1491,7 +1491,7 @@ def apply_constraint(arg, fx_arg):
 make_fallback(aten.upsample_bicubic2d_backward, require_contiguous)
 
-make_fallback(aten.view_as_complex.default, require_contiguous)
+make_fallback(aten.view_as_complex, require_contiguous)
```
We shouldn't need this change to make it work, they are equivalent

int3 replied:
Without that change we get

```
LoweringException: AssertionError: view_as_complex.default is not supported with cpp wrapper
```

desertfire replied:
It's being fixed in https://github.com/pytorch/pytorch/issues/102752. Let's not duplicate work.

eellison reviewed (approved):
Looks good but we should make the overloaded registration work too

int3 commented:
@int3 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D46585029).

int3 commented:
@pytorchbot merge

int3 committed:
Update on "Remove redundant fallback for view_as_complex"


This enables lowering to work for it

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 aakhundov

Differential Revision: [D46585029](https://our.internmc.facebook.com/intern/diff/D46585029)

[ghstack-poisoned]

int3 commented:
@int3 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D46585029).

int3 commented:
@pytorchbot merge

int3 commented:
@pytorchbot merge

int3 commented:
@pytorchbot label "topic: not user facing"

int3 commented:
@pytorchbot merge
